### PR TITLE
APPS-1338 Skip not found error on files streaming from directory

### PR DESCRIPTION
**Pull request by filkeith**

### CODE REVIEW

File: io/azure/blob/reader.go [hunk 1]
```diff
@@ -138,7 +141,15 @@ func (r *Reader) streamDirectory(
...
+				if errors.As(err, &respErr) && respErr.StatusCode == http.StatusNotFound {
+					continue
+				}
+
+				errorsCh <- fmt.Errorf("failed to create reader from directory file %s: %w", *blob.Name, err)
+
+				return
 			}
 
 			readersCh <- resp.Body
```
[the remark this answers is not in the document]
Comment (filkeith) [reply io/azure/blob/reader.go:155]: In this case resp in not pointer it is  `azblob.DownloadStreamResponse`

**Review by korotkov-aerospike — APPROVED** [review]

File: io/aws/s3/reader.go [hunk 2]
```diff
@@ -156,12 +160,26 @@ func (r *Reader) streamDirectory(
 				Key:    p.Key,
 			})
 			if err != nil {
+				// Skip 404 not found error.
+				var opErr *smithy.OperationError
+				if errors.As(err, &opErr) {
+					var httpErr *awsHttp.ResponseError
+					if errors.As(opErr.Err, &httpErr) {
+						if httpErr.HTTPStatusCode() == http.StatusNotFound {
```
[the remark this answers is not in the document]
Comment (korotkov-aerospike) [reply io/aws/s3/reader.go:168]: or even just extract boolean predicate isNotFoundError(err)

[the remark this answers is not in the document]
Comment (filkeith) [reply io/aws/s3/reader.go:168]: fixed

File: io/azure/blob/reader.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (reugn) [reply io/azure/blob/reader.go:155]: Is there a check for it being empty?

[the remark this answers is not in the document]
Comment (filkeith) [reply io/azure/blob/reader.go]: we can check body itself

**Review by reugn — APPROVED** [review]

**Pull request merged**